treeherder/model/tasks.py

```diff
@@ -11,19 +11,37 @@
 from treeherder.model.pulse_publisher import load_schemas
 
 # Load schemas for validation of messages published on pulse
-source_folder = os.path.dirname(os.path.realpath(__file__))
-schema_folder = os.path.join(source_folder, '..', '..', 'schemas')
-schemas = load_schemas(schema_folder)
+SOURCE_FOLDER = os.path.dirname(os.path.realpath(__file__))
+SCHEMA_FOLDER = os.path.join(SOURCE_FOLDER, '..', '..', 'schemas')
+PULSE_SCHEMAS = load_schemas(SCHEMA_FOLDER)
 
-# Create publisher, if username and password is present
-publisher = None
-if settings.PULSE_EXCHANGE_NAMESPACE:
-    publisher = TreeherderPublisher(
-        namespace=settings.PULSE_EXCHANGE_NAMESPACE,
-        uri=settings.PULSE_URI,
-        schemas=schemas
-    )
+class LazyPublisher():
+    """
+    Singleton for lazily connecting to the pulse publisher.
+    """
+
+    def __init__(self):
+        self.publisher = False
+
+    def get_publisher(self):
+        """
+        Attempt to get the publisher.
+        """
+
+        if self.publisher is not False:
+            return self.publisher;
+
+        # Create publisher, if username and password is present
+        if settings.PULSE_EXCHANGE_NAMESPACE:
+            self.publisher = TreeherderPublisher(
+                namespace=settings.PULSE_EXCHANGE_NAMESPACE,
+                uri=settings.PULSE_URI,
+                schemas=PULSE_SCHEMAS
+            )
+        else:
+            self.publisher = None
```

Comment:
We need a final |return self.publisher here otherwise this won't work on the first call, right?